src/PerfView.Tests/EventViewerTests.cs

```diff
@@ -0,0 +1,290 @@
+                {
+                    Viewer.Focus();
+                    doAfter?.Invoke();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Test EventSource implementation with problematic EventNames for XML escaping testing
+        /// </summary>
+        private class XmlEscapeTestEventSource : EventSource
+        {
+            private readonly EventRecord[] _events;
+
+            public XmlEscapeTestEventSource()
+            {
+                _events = new EventRecord[]
+                {
+                    // The main test case from issue #927
+                    new TestEventRecord("Enter\" providername=\"Microsoft-Azure-Devices", "Process(3164)", 783264.803),
+                    
+                    // Additional test cases for various XML special characters
+                    new TestEventRecord("<script>alert('xss')</script>", "Process(1234)", 1000.0),
+                    new TestEventRecord("Test & Company", "Process(5678)", 2000.0),
+                    new TestEventRecord("Quote: \"Hello\"", "Process(9012)", 3000.0),
+                    new TestEventRecord("Apostrophe: 'Hello'", "Process(3456)", 4000.0),
+                };
+                
+                MaxEventTimeRelativeMsec = double.PositiveInfinity;
+            }
+
+            public override void ForEach(Func<EventRecord, bool> callback)
+            {
+                foreach (var eventRecord in _events)
+                {
+                    if (!callback(eventRecord))
+                        break;
+                }
+            }
+
+            public override void SetEventFilter(List<string> eventNames)
+            {
+                // Not needed for this test
+            }
+
+            public override ICollection<string> EventNames => 
+                new List<string> { "TestEvent1", "TestEvent2", "TestEvent3", "TestEvent4", "TestEvent5" };
+
+            public override EventSource Clone()
+            {
+                return new XmlEscapeTestEventSource();
+            }
+        }
+
+        /// <summary>
+        /// Test implementation of EventRecord for testing XML escaping
+        /// </summary>
+        private class TestEventRecord : EventRecord
+        {
+            private readonly string _eventName;
+            private readonly string _processName;
+            private readonly double _timeStamp;
+
+            public TestEventRecord(string eventName, string processName, double timeStamp) : base()
+            {
+                _eventName = eventName;
+                _processName = processName;
+                _timeStamp = timeStamp;
+            }
+
+            public override string EventName => _eventName;
+            public override string ProcessName => _processName;
+            public override double TimeStampRelatveMSec => _timeStamp;
+            public override List<Payload> Payloads => new List<Payload>();
+            public override string Rest => "";
```

Comment:
Part of this change also fixes the quote escaping behavior for `Rest`.  Please also test `Rest` with and without values that require quote escaping.